src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -164,8 +164,17 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 		      && (_vehicle_local_position_setpoint.vz >= land_speed_threshold);
 	bool hit_ground = _in_descend && !vertical_movement;
 
+	bool ground_contact = false;
+
+	if (_maybe_landed_hysteresis.get_state() || _landed_hysteresis.get_state()) {
```

Comment:
@BazookaJoe1900 Your concern is proof for me that it complicates things. I had to think through twice but this condition is true when ground conact and maybe landed were already detected and we are checking if the conditions for ground contact are still holding. I consider it bad practice to suddenly change the condition at a later stage.

Reply:
@MaEtUgR @BazookaJoe1900 This condition could also be left away, it's not the core part.
The main part is the fact that we can only trigger ground contact if both low_thrust and hit_ground are true.
`The hit ground condition was originally added to accelerate landing detection for drones that have very conservative altitude control tuning and would therefore take very long to go low on thrust when not achieving their downwards velocity setpoint because they arrived on the ground`
@MaEtUgR For this reason I once added LNDMC_LOW_T_THR to be able to fine tune the low thrust setting.

As stated in the description of the PR: I'm not sure if this change makes sense for the community.
These changes work very well when you have your parameters setup properly and it's practically highly unlikely to get false positives.
The current land detector produces false positives and they are not very unlikely to happen.
But most vehicles, especially the smaller ones can easily recover and all you hear is a sudden surge in throttle.